python/ambassador/ir/irhost.py

```diff
@@ -167,3 +160,41 @@ def load_all(cls, ir: 'IR', aconf: Config) -> None:
                     ir.save_host(host)
                 else:
                     ir.logger.info(f"HostFactory: not saving inactive host {host.pretty()}")
+
+        if ir.edge_stack_allowed:
+            # We're running Edge Stack. Figure out how many hosts we have, and whether
+            # we have any termination contexts.
+            host_count = len(ir.get_hosts() or [])
+            contexts = ir.get_tls_contexts() or []
+
+            found_termination_context = False
+            for ctx in contexts:
+                if ctx.get('hosts'):  # not None and not the empty list
+                    found_termination_context = True
+
+            ir.logger.info(f"HostFactory: FTC {found_termination_context}, host_count {host_count}")
```

Comment:
I don't like that name.  Not enough buzz-words.  Try to fit in "bean" or "monoid" or "functor".

Reply:
`HostBeanFactoryMonoidFunctorFactory`?

Reply:
Exactly!